tests/Tests/ORM/Functional/Ticket/GH11500Test.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\PersistentCollection;
+use Doctrine\ORM\Persisters\Collection\OneToManyPersister;
+use Doctrine\Tests\Mocks\ConnectionMock;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function array_pop;
+use function assert;
+
+class GH11500Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH11500AbstractTestEntity::class,
+            GH11500TestEntity::class,
+            GH11500TestEntityHolder::class,
+        ]);
+    }
+
+    public function testDeleteOneToManyCollectionWithSingleTableInheritance(): void
+    {
+        $em   = $this->getTestEntityManager();
+        $conn = $em->getConnection();
+        assert($conn instanceof ConnectionMock);
+
+        $testEntity       = new GH11500TestEntity();
+        $testEntityHolder = new GH11500TestEntityHolder();
+
+        $testEntity->testEntityHolder = $testEntityHolder;
+        $testEntityHolder->testEntities->add($testEntity);
+
+        $em->persist($testEntity);
+        $em->persist($testEntityHolder);
+        $em->flush();
+
+        assert($testEntityHolder->testEntities instanceof PersistentCollection);
+
+        $persister = new OneToManyPersister($em);
+        $persister->delete($testEntityHolder->testEntities);
+
+        $updates       = $conn->getExecuteStatements();
+        $lastStatement = array_pop($updates);
+
+        self::assertEquals('DELETE FROM one_to_many_single_table_inheritance_test_entities WHERE test_entity_holder_id = ? AND type = ?', $lastStatement['sql']);
+        self::assertEquals('test_entity', $lastStatement['params'][1] ?? null);
+    }
+}
+
+
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="one_to_many_single_table_inheritance_test_entities")
+ * @ORM\InheritanceType("SINGLE_TABLE")
+ * @ORM\DiscriminatorColumn(name="type", type="string")
+ * @ORM\DiscriminatorMap({"test_entity"="GH11500TestEntity"})
+ */
+class GH11500AbstractTestEntity
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue
+     */
+    public int $id;
```

Comment:
You must write code that is compatible with PHP 7.1, sorry.

Reply:
Ah, my apologies. Have amended.